devices/flash-zynq7000/flashdrv.c

```diff
@@ -187,22 +187,48 @@ static int flashdrv_welSet(unsigned int cmdID)
 }
 
 
-static int flashdrv_sectorErase(addr_t offs)
+static int flashdrv_chipErase(void)
 {
-	u32 id = 0;
 	ssize_t res;
 	time_t timeout;
-	const flash_cmd_t *cmd;
+
 	const flash_cfi_t *cfi = &fdrv_common.info.cfi;
+	const flash_cmd_t *cmd = &fdrv_common.info.cmds[flash_cmd_be];
 
-	if (offs > CFI_SIZE_FLASH(cfi->chipSize))
-		return -EINVAL;
+	res = flashdrv_welSet(flash_cmd_wren);
+	if (res < 0) {
+		return res;
+	}
 
-	res = flashdrv_regionFind(offs, &id);
-	if (res < 0)
+	hal_memset(fdrv_common.cmdTx, 0, cmd->size);
+	fdrv_common.cmdTx[0] = cmd->opCode;
+
+	qspi_start();
+	res = qspi_polledTransfer(fdrv_common.cmdTx, NULL, cmd->size, TIMEOUT_CMD_MS);
+	qspi_stop();
+
+	if (res < 0) {
 		return res;
+	}
+
+	timeout = CFI_TIMEOUT_MAX_ERASE(cfi->timeoutTypical.chipErase, cfi->timeoutMax.chipErase) + TIMEOUT_CMD_MS;
+
+	return flashdrv_wipCheck(timeout);
+}
 
-	switch (CFI_SIZE_SECTION(cfi->regs[id].size)) {
+
+static int flashdrv_sectorErase(addr_t offs, size_t sectSz)
+{
+	ssize_t res;
```

Comment:
The result of the `flashdrv_sectorErase` function is unlikely to be of type of `ssize_t`, here only `qspi_polledTransfer()` result is of `ssize_t` type, the other returns `int`, the same for `flashdrv_chipErase(void)`.
```suggestion
	int res;
```
if so, then
```diff
-  res = qspi_polledTransfer(fdrv_common.cmdTx, fdrv_common.cmdRx, cmd->size, TIMEOUT_CMD_MS);
+  res = (int)qspi_polledTransfer(fdrv_common.cmdTx, fdrv_common.cmdRx, cmd->size, TIMEOUT_CMD_MS);TIMEOUT_CMD_MS);
```
because only negative values or zero that fall within `int` range are of interest to us, and number of erased sectors or erased size is not returned by the function.

Reply:
Done